app/internal/import_file.py

```diff
@@ -0,0 +1,185 @@
+import datetime
+import os
+from pathlib import Path
+import re
+from typing import Any, Dict, List, Tuple, Union
+
+from icalendar import Calendar
+
+from app.database.models import Event
+from app.database.database import SessionLocal
+
+
+NUM_OF_VALUES = 4  # Event contains head, content, start_date and end_date.
+MAX_FILE_SIZE_MB = 5  # 5MB
+VALID_FILE_EXTENSION = (".txt", ".csv", ".ics")  # Can import only these files.
+VALID_YEARS = 20  # Events must be within 20 years range from the current year.
+EVENT_HEADER_NOT_EMPTY = 1  # 1- for not empty, 0- for empty.
+EVENT_HEADER_LIMIT = 50  # Max characters for event header.
+EVENT_CONTENT_LIMIT = 500  # Max characters for event characters.
+MAX_EVENTS_START_DATE = 10  # Max Events with the same start date.
+
+
+def is_file_size_valid(file: str, max_size: int = MAX_FILE_SIZE_MB) -> bool:
+    file_size = os.stat(file).st_size / 1048576  # convert bytes to MB.
+    return file_size <= max_size
+
+
+def is_file_extension_valid(file: str,
+                            extension: Union[str, Tuple[str, ...]]
+                            = VALID_FILE_EXTENSION) -> bool:
+    return file.lower().endswith(extension)
+
+
+def is_file_exist(file: str) -> bool:
+    if Path(file).is_file():
+        return True
+    return False
+
+
+def is_date_in_range(date: Union[str, datetime.datetime],
+                     valid_dates: int = VALID_YEARS) -> bool:
+    """
+    check if date is valid and in the range according to the rule we have set
+    """
+    now_year = datetime.datetime.now().year
+    if isinstance(date, str):
+        try:
+            check_date = datetime.datetime.strptime(date, "%m-%d-%Y")
+        except ValueError:
+            return False
+    else:
+        check_date = date
+    if check_date.year > now_year + valid_dates or \
+       check_date.year < now_year - valid_dates:
+        return False
+    return True
+
+
+def is_event_text_valid(row: str) -> bool:
+    """Check if the row contains valid data"""
+    get_values = re.findall(r"^(\w{" + str(EVENT_HEADER_NOT_EMPTY) + "," +
```

Comment:
Prefer compiling the regular expression using `re.compile` and  giving it a name as a constant variable at the top :)

Reply:
Done